addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -379,3 +382,255 @@ def _l10n_es_edi_facturae_render_facturae(self):
+
+        if file_data['type'] == 'xml' and is_facturae(file_data['xml_tree']):
+            return self._import_invoice_facturae
+
+        return super()._get_edi_decoder(file_data, new=new)
+
+    def _import_invoice_facturae(self, invoice, file_data, new=False):
+        tree = file_data['xml_tree']
+        is_bill = invoice.move_type.startswith('in')
+        partner = self._import_get_partner(tree, is_bill)
+        self._import_invoice_facturae_invoices(invoice, partner, tree)
+
+    def _import_get_partner(self, tree, is_bill):
+        # If we're dealing with a vendor bill, then the partner is the seller party, if an invoice then it's the buyer.
+        party = tree.xpath('//SellerParty/*[2]') if is_bill else tree.xpath('//BuyerParty/*[2]')
+        if party:
+            partner_vals = self._import_extract_partner_values(party[0])
+            return self._import_create_or_retrieve_partner(partner_vals)
+        return None
+
+    def _import_extract_partner_values(self, party_node):
+        name = find_xml_value('.//CorporateName|.//Name', party_node)
+        first_surname = find_xml_value('.//FirstSurname', party_node)
+        second_surname = find_xml_value('.//SecondSurname', party_node)
+        phone = find_xml_value('.//Telephone', party_node)
+        mail = find_xml_value('.//ElectronicMail', party_node)
+        country_code = find_xml_value('.//CountryCode', party_node)
+        vat = find_xml_value('.//TaxIdentificationNumber', party_node)
+
+        full_name_parts = [part for part in [name, first_surname, second_surname] if part is not None]
+        full_name = ' '.join(full_name_parts)
+
+        return {'name': full_name, 'vat': vat, 'phone': phone, 'email': mail, 'country_code': country_code}
+
+    def _import_create_or_retrieve_partner(self, partner_vals):
+        name = partner_vals['name']
+        vat = partner_vals['vat']
+        phone = partner_vals['phone']
+        mail = partner_vals['email']
+        country_code = partner_vals['country_code']
+
+        partner = self.env['res.partner']._retrieve_partner(name=name, vat=vat, phone=phone, mail=mail)
+
+        if not partner and name:
+            partner_vals = {'name': name, 'email': mail, 'phone': phone}
+            country = self.env.ref(f'base.{country_code.lower()}', raise_if_not_found=False) if country_code else False
+            if country:
+                partner_vals['country_id'] = country.id
+            partner = self.env['res.partner'].create(partner_vals)
+            if vat and self.env['res.partner']._run_vat_test(vat, country):
+                partner.vat = vat
+
+        return partner
+
+    def _import_invoice_facturae_invoices(self, invoice, partner, tree):
+        invoices = tree.xpath('//Invoice')
+        if not invoices:
+            return
+
+        self._import_invoice_facturae_invoice(invoice, partner, invoices[0])
+
+        # There might be other invoices inside the facturae.
+        for node in invoices[1:]:
```

Comment:
I think this a great idea but maybe it should be done in another task since it'll require some rewriting for the other edi modules as well. This task has already been hindered a long time by the xml signature stuff so I say we push this one and do this in another task, what do you think? :)